cli/lib/nf3.mjs

```diff
@@ -255,35 +261,64 @@ class Nf3 {
   }
 
   /**
-  Method for signing and submitting an Ethereum transaction to the
+  Method for signing an Ethereum transaction to the
   blockchain.
   @method
   @async
   @param {object} unsignedTransaction - An unsigned web3js transaction object.
   @param {string} shieldContractAddress - The address of the Nightfall_3 shield address.
   @param {number} fee - the value of the transaction.
   This can be found using the getContractAddress convenience function.
+  @returns {Oject} Signed transaction.
+  */
+  async _signTransaction(unsignedTransaction, contractAddress, fee) {
+    let tx;
+    let signed;
+
+    await this.nonceMutex.runExclusive(async () => {
+      // estimate the gasPrice
+      const gasPrice = await this.estimateGasPrice();
+      // Estimate the gasLimit
+      const gas = await this.estimateGas(contractAddress, unsignedTransaction);
+
+      // if we don't have a nonce, we must get one from the ethereum client
+      if (!this.nonce) this.nonce = await this.web3.eth.getTransactionCount(this.ethereumAddress);
+
+      tx = {
+        from: this.ethereumAddress,
+        to: contractAddress,
+        data: unsignedTransaction,
+        value: fee,
+        gas,
+        gasPrice,
+        nonce: this.nonce,
+      };
+      this.nonce++;
+
+      if (this.ethereumSigningKey) {
```

Comment:
As the `gasPrice/estimateGas` methods, this could be moved outside the mutex too, the `tx` variable is safe to be used.